src/platform/utilities/toggles/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import environments from 'platform/utilities/environment';
+
+function makeEnvironmentToggleValues(env = environments) {
+  return {
+    develop: env.isDev(),
+    localhost: env.isLocalhost(),
+    notProduction: !env.isProduction(),
+    staging: env.isStaging(),
+    production: env.isProduction(),
+  };
+}
+
+const ToggleContext = React.createContext();
+
+function ToggleProvider(props) {
+  const {
+    children,
+    initialToggleValues,
+    subscribeToUpdateToggles = () => {},
+    unsubscribeToUpdateToggles = () => {},
+  } = props;
+
+  const [toggleValues, setToggleValues] = useState(initialToggleValues);
+
+  function handleToggleValuesUpdate(newToggleValues) {
+    setToggleValues(newToggleValues);
+  }
+
+  useEffect(
+    () => {
+      subscribeToUpdateToggles(handleToggleValuesUpdate);
+
+      return () => unsubscribeToUpdateToggles();
+    },
+    [toggleValues, subscribeToUpdateToggles, unsubscribeToUpdateToggles],
+  );
+
+  return (
+    <ToggleContext.Provider value={toggleValues}>
+      {children}
+    </ToggleContext.Provider>
+  );
+}
+
+const initialToggleValues = makeEnvironmentToggleValues();
+
+const withToggleProvider = (
+  WrappedComponent,
+  toggleValues = initialToggleValues,
+) => props => (
+  <ToggleProvider initialToggleValues={toggleValues}>
+    <WrappedComponent {...props} />
+  </ToggleProvider>
+);
+
+const Toggle = ({ children, ...props }) => {
+  const toggles = Object.keys(props);
+
+  return (
+    <ToggleContext.Consumer>
+      {toggleValues => {
+        const showChildren = toggles.find(toggle => {
```

Comment:
You can use this like 
```
<Toggle notProduction>
  <span>some stuff</span>
</Toggle>
```

The `<Toggle />` will search its own props against the `toggleValues` provided by the `ToggleContext.Consumer`.